packages/headless/src/features/insight-search/insight-search-analytics-actions.ts

```diff
@@ -25,3 +25,38 @@ export const logQueryError = (error: SearchAPIErrorWithStatusCode) =>
         errorMessage: error.message,
       })
   )();
+
+export const logContextChanged = (caseId: string, caseNumber: string) =>
+  makeInsightAnalyticsAction(
+    'analytics/contextChanged',
+    AnalyticsType.Search,
+    (client, state) => {
+      const meta = {
+        caseId,
+        caseNumber,
+        caseContext: state.insightCaseContext?.caseContext || {},
+      };
+      client.logContextChanged(meta);
```

Comment:
Adding some tests just to validate that the right data is passed down to the client would be nice.

Reply:
tests added👌